litefs/disaster-recovery.html.md.erb

```diff
@@ -0,0 +1,174 @@
+```
+
+If you're running on the Fly platform, you can see the logs with:
+
+```sh
+fly logs
+```
+
+Note the cluster id specified at the end of the log message: `LFSC25AD000F32ED02A9` in
+our example. Copy this value to use later.
+
+## Update the cluster id
+
+Now that you have the correct cluster id, you can connect to your primary node and
+update the cluster id file. If your app is deployed on the Fly Platform, you can
+connect with:
+
+```sh
+fly ssh console
+```
+
+Then update the `clusterid` file:
+
+```sh
+echo LFSC25AD000F32ED02A9 > /var/lib/litefs/clusterid
+```
+
+After this, you should restart litefs. If you're using the Fly Platform, close
+your ssh session, and then restart your app:
+
+```sh
+fly app restart
+```
+
+Check your logs again, and confirm you don't see the error message anymore!
+You should also take a look at your app and confirm you see the data you expect
+before you continue.
+
+## Resolve consul key error
+
+If you're running on the Fly Platform (or you're using Consul for lease management
+elsewhere), you may see the following errors in your logs:
+
+```
+2023-07-10T13:30:25Z app[784e900c2de378] ams [info]level=INFO msg="cannot connect, \"consul\" lease already initialized with different ID: LFSCA5E3680C1C2FCFFE"
+```
+
+You have two options to resolve this:
+
+- change the Consul key value that LiteFS uses. This is easy, but somewhat hacky/ugly.
+- remove the "wrong" key from Consul. This is more difficult but cleaner.
+
+### Easy option: change the Consul key
+
+You can simply update the `lease.consul.key` value in the `litefs.yml` file. The old one will
+still exist in Consul, but LiteFS won't care. Here's how to update your `litefs.yml`:
+
+```yml
+lease:
+  type: consul
+  ...
+  consul:
+    url: "${FLY_CONSUL_URL}"
+    key: "litefs/${FLY_APP_NAME}-v2" # added '-v2'
+```
+
+If you're using the Fly Platform, redeploy your app:
+
+```sh
+fly deploy
+```
+
+### Cleaner option: remove the wrong key from Consul
+
+First, you'll need to add `consul` in your image. You can do this in your `Dockerfile` and re-deploy,
+or just connect to your primary node via ssh and install it there. Here's how to install it:
+
+```sh
+# For debian/ubuntu-based images
+apt-get install consul
+# For alpine-based images
+apk add consul
+```
+
+Then, take a look at your `FLY_CONSUL_URL`:
+
+```sh
+echo $FLY_CONSUL_URL
+```
+
+This url is in the format `https://:{TOKEN}@{HOST}/{PREFIX}`. You'll need each of these values
+separately. You'll also need the value of `lease.consul.key` from your `litefs.yml` file.
+
+Then you can delete the key with:
+
+```sh
+export TOKEN=(copy the token you found from your consul url)
+export HOST=(copy the host you found from your consul url)
+export PREFIX=(copy the prefix you found from your consul url)
+export LITEFS_CONSUL_KEY=(copy the key from your litefs.yml file)
+
+# and then delete the key!
+consul kv delete -http-addr=https://$HOST -token=$TOKEN $PREFIX/$LITEFS_CONSUL_KEY/clusterid
+```
```

Comment:
It might be worth mentioning that they can use `consul kv export` to list out their keys so they can find the one they're looking for.

```
consul kv export -http-addr=https://$HOST -token=$TOKEN
```

Reply:
I tried to add this in, but it got confusing. I _think_ it's better to leave it as simple/straight path as possible. As a compromise though I can link [the consul kv CLI docs](https://developer.hashicorp.com/consul/commands/kv).